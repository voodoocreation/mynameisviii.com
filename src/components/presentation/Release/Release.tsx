import * as React from "react";
import {
  MdAccessTime,
  MdAlbum,
  MdDateRange,
  MdFormatListNumbered,
  MdMusicNote
} from "react-icons/md";
import {
  FormattedDate,
  FormattedMessage,
  injectIntl,
  WrappedComponentProps
} from "react-intl";

import { IRelease } from "../../../models/root.models";
import Schema from "../../schema/Release";
import Carousel from "../Carousel/Carousel";
import DateTime from "../DateTime/DateTime";
import Image from "../Image/Image";
import Link from "../Link/Link";
import Meta from "../Meta/Meta";
import MetaBar from "../MetaBar/MetaBar";
import PageHeader from "../PageHeader/PageHeader";
import PlatformIcon from "../PlatformIcon/PlatformIcon";

import "./Release.scss";

interface IProps extends IRelease, WrappedComponentProps {
  onCarouselSlideChange?: (index: number) => void;
}

class Release extends React.Component<IProps> {
  public render() {
    const { intl, onCarouselSlideChange, ...release } = this.props;

    return (
      <article className="Release">
        <PageHeader>{release.title}</PageHeader>

        {this.renderMetaSection()}

        <div className="Release--body">
          {this.renderImagesSection()}

          <div className="Release--details">
            <section
              className="Release--description"
              dangerouslySetInnerHTML={{ __html: release.description }}
            />

            {this.renderTracklist()}
            {this.renderStreamList()}
            {this.renderBuyList()}

            <section className="Release--copyright">
              <p>
                ℗© <FormattedDate value={release.releasedOn} year="numeric" />{" "}
                {release.recordLabel}
              </p>
            </section>
          </div>
        </div>

        <Schema {...this.props} />
      </article>
    );
  }

  private renderMetaSection = () => {
    const { genre, length, releasedOn, tracklist, type } = this.props;
    const trackCount = tracklist.reduce((acc, curr) => acc + curr.length, 0);

    return (
      <MetaBar className="Release--meta">
        <Meta className="Release--type" icon={<MdAlbum />} labelIntlId="TYPE">
          <FormattedMessage id={type.toUpperCase()} values={{ count: 1 }} />
        </Meta>

        <Meta
          className="Release--releasedOn"
          icon={<MdDateRange />}
          labelIntlId="RELEASED"
        >
          <DateTime
            isDateOnly={true}
            isRelative={false}
            options={{
              day: "numeric",
              month: "long",
              year: "numeric"
            }}
            value={releasedOn}
          />
        </Meta>

        <Meta
          className="Release--length"
          icon={<MdAccessTime />}
          labelIntlId="LENGTH"
        >
          {length}
        </Meta>

        <Meta
          className="Release--genre"
          icon={<MdMusicNote />}
          labelIntlId="GENRE"
        >
          {genre}
        </Meta>

        <Meta className="Release--tracks" icon={<MdFormatListNumbered />}>
          <FormattedMessage id="X_TRACKS" values={{ trackCount }} />
        </Meta>
      </MetaBar>
    );
  };

  private renderImagesSection = () => (
    <section className="Release--images">
      <Carousel onSlideChange={this.onCarouselSlideChange}>
        {this.props.images.map((image, index) => (
          <Image
            alt={image.title}
            caption={image.title}
            className="Release--images--image"
            key={`image-${index}`}
            src={image.imageUrl}
          />
        ))}
      </Carousel>
    </section>
  );

  private renderTracklist = () => (
    <section className="Release--tracklist">
      <h2>
        <FormattedMessage id="TRACKLIST" />
      </h2>

      {this.props.tracklist.map((album, albumIndex) => (
        <React.Fragment key={albumIndex}>
          {this.props.tracklist.length > 1 ? (
            <h3>
              <FormattedMessage
                id="DISC_NUMBER"
                values={{ number: albumIndex }}
              />
            </h3>
          ) : null}
          <ol>
            {album.map(track => (
              <li key={track.title}>
                <span className="Release--tracklist--title">{track.title}</span>{" "}
                <span className="Release--tracklist--genre">{track.genre}</span>{" "}
                <span className="Release--tracklist--length">
                  {track.length}
                </span>
              </li>
            ))}
          </ol>
        </React.Fragment>
      ))}
    </section>
  );

  private renderStreamList = () =>
    this.props.streamList.length > 0 ? (
      <section className="Release--streamList">
        <h2>
          <FormattedMessage id="STREAM" />
        </h2>

        <ul className="Release--platformList">
          {this.props.streamList.map(link => (
            <li key={link.platform}>
              <Link
                href={link.url}
                isExternal={true}
                title={this.props.intl.formatMessage(
                  {
                    id: `STREAM_ON_PLATFORM`
                  },
                  {
                    platform: this.props.intl.formatMessage({
                      id: link.platform.toUpperCase()
                    }),
                    title: this.props.title
                  }
                )}
              >
                <PlatformIcon value={link.platform} />{" "}
                <FormattedMessage id={link.platform.toUpperCase()} />
              </Link>
            </li>
          ))}
        </ul>
      </section>
    ) : null;

  private renderBuyList = () =>
    this.props.buyList.length > 0 ? (
      <section className="Release--buyList">
        <h2>
          <FormattedMessage id="BUY" />
        </h2>

        <ul className="Release--platformList">
          {this.props.buyList.map(link => (
            <li key={link.platform}>
              <Link
                href={link.url}
                isExternal={true}
                title={this.props.intl.formatMessage(
                  {
                    id: `BUY_FROM_PLATFORM`
                  },
                  {
                    platform: this.props.intl.formatMessage({
                      id: link.platform.toUpperCase()
                    }),
                    title: this.props.title
                  }
                )}
              >
                <PlatformIcon value={link.platform} />{" "}
                <FormattedMessage id={link.platform.toUpperCase()} />
              </Link>
            </li>
          ))}
        </ul>
      </section>
    ) : null;

  private onCarouselSlideChange = (index: number) => {
    if (this.props.onCarouselSlideChange) {
      this.props.onCarouselSlideChange(index);
    }
  };
}

export default injectIntl(Release);
