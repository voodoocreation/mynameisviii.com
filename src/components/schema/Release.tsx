import * as React from "react";
import stripTags from "striptags";

import { absoluteUrl, lengthToDuration } from "../../helpers/dataTransformers";
import { IRelease, IReleaseTrack } from "../../models/root.models";
import Schema from "./Schema";

const getFlatTracklist = (tracklist: IReleaseTrack[][]) =>
  tracklist.reduce((acc, curr) => [...acc, ...curr], []);

const Release: React.FC<IRelease> = props => (
  <Schema
    {...{
      "@id": absoluteUrl(`/releases/${props.slug}`),
      "@type": "MusicAlbum",
      albumProductionType: `http://schema.org/${props.productionType}`,
      albumRelease: {
        "@type": "MusicRelease",
        duration: lengthToDuration(props.length),
        name: props.title,
        recordLabel: props.recordLabel
      },
      albumReleaseType: `http://schema.org/${props.type}`,
      byArtist: {
        "@type": "MusicGroup",
        name: props.artist.name
      },
      datePublished: props.releasedOn,
      description: stripTags(props.description).replace(/\n/g, ""),
      image: props.images[0].imageUrl,
      mainEntityOfPage: {
        "@id": absoluteUrl(`/releases/${props.slug}`)
      },
      name: props.title,
      track: {
        "@type": "ItemList",
        itemListElement: getFlatTracklist(props.tracklist).map(
          (track, index) => ({
            "@type": "ListItem",
            item: {
              "@type": "MusicRecording",
              duration: lengthToDuration(track.length),
              name: track.title,
              recordingOf: {
                "@type": "MusicComposition",
                name: track.title
              }
            },
            position: index + 1
          })
        ),
        numberOfItems: getFlatTracklist(props.tracklist).length
      }
    }}
  />
);

export default Release;
