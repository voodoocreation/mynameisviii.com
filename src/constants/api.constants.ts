export enum BOOLEAN {
  TRUE = "y",
  FALSE = "n",
}
