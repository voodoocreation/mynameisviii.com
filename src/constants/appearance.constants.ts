export enum STATUS {
  CANCELLED = "EventCancelled",
  POSTPONED = "EventPostponed",
  RESCHEDULED = "EventRescheduled",
  SCHEDULED = "EventScheduled"
}

export enum TYPE {
  ART = "VisualArtsEvent",
  EXHIBITION = "ExhibitionEvent",
  FESTIVAL = "Festival",
  MUSIC = "MusicEvent",
  SCREENING = "ScreeningEvent",
  SOCIAL = "SocialEvent"
}
