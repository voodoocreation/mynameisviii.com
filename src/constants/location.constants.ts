export enum TYPE {
  CITY = "City",
  EVENT_VENUE = "EventVenue",
  MOVIE_THEATER = "MovieTheater",
  MUSIC_VENUE = "MusicVenue",
  PARK = "Park",
  PERFORMING_ARTS_THEATER = "PerformingArtsTheater",
  STADIUM = "StadiumOrArena",
}
