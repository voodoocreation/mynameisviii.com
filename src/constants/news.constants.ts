export enum TYPE {
  APPEARANCE = "appearance",
  NEWS = "news",
  RELEASE = "release"
}
