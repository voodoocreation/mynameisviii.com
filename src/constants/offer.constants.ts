export enum AVAILABILITY {
  IN_STOCK = "InStock",
  IN_STORE = "InStoreOnly",
  LIMITED = "LimitedAvailability",
  OUT_OF_STOCK = "OutOfStock",
  ONLINE = "OnlineOnly",
  PRE_ORDER = "PreOrder",
  PRE_SALE = "PreSale",
  SOLD_OUT = "SoldOut",
}
