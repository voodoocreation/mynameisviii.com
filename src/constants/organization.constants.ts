export enum TYPE {
  CORPORATION = "Corporation",
  CHARITY = "FundingScheme",
  LOCAL_BUSINESS = "LocalBusiness",
  MEDIA = "NewsMediaOrganization",
  PERFORMER = "PerformingGroup",
  PROJECT = "Project",
}
