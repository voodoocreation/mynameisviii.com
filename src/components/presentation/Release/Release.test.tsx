import { PLATFORM } from "../../../constants/release.constants";
import {
  image,
  release,
  releasePlatformLink,
  releaseTrack
} from "../../../models/root.models";
import ComponentTester from "../../../utilities/ComponentTester";
import Release from "./Release";

const component = new ComponentTester(Release).withDefaultProps(
  release({
    artist: {
      name: "Artist Name",
      url: "URL"
    },
    buyList: [],
    description: "Description",
    genre: "Genre",
    images: [
      {
        imageUrl: "Image URL",
        title: "Image title"
      }
    ],
    length: "5:00",
    recordLabel: "Record Label",
    releasedOn: "2017-01-01T00:00:00",
    slug: "release-1",
    streamList: [],
    title: "Title",
    tracklist: [
      [
        {
          genre: "Genre",
          length: "5:00",
          title: "Title",
          url: "URL"
        }
      ]
    ]
  })
);

describe("[presentation] <Release />", () => {
  describe("when the minimum props are defined, with only one disc", () => {
    const { wrapper } = component.render();

    it("doesn't render the disc number heading", () => {
      expect(wrapper.find(".Release--tracklist h3")).toHaveLength(0);
    });

    it("doesn't render the stream list section", () => {
      expect(wrapper.find(".Release--streamList")).toHaveLength(0);
    });

    it("doesn't render the buy list section", () => {
      expect(wrapper.find(".Release--buyList")).toHaveLength(0);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when all props are defined, with multiple discs", () => {
    const { wrapper } = component
      .withProps({
        buyList: [
          releasePlatformLink({ platform: PLATFORM.ITUNES, url: "URL" })
        ],
        streamList: [
          releasePlatformLink({ platform: PLATFORM.SPOTIFY, url: "URL" })
        ],
        tracklist: [
          [
            releaseTrack({ title: "Disc 1, Track 1" }),
            releaseTrack({ title: "Disc 1, Track 2" })
          ],
          [
            releaseTrack({ title: "Disc 2, Track 1" }),
            releaseTrack({ title: "Disc 2, Track 2" })
          ]
        ]
      })
      .render();

    it("renders the disc number headings", () => {
      expect(wrapper.find(".Release--tracklist h3")).toHaveLength(2);
    });

    it("renders the stream list section", () => {
      expect(wrapper.find(".Release--streamList")).toHaveLength(1);
    });

    it("renders the buy list section", () => {
      expect(wrapper.find(".Release--buyList")).toHaveLength(1);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when interacting with the carousel", () => {
    const { props, wrapper } = component
      .withProps({
        images: [
          image({ imageUrl: "Image URL", title: "Image 1" }),
          image({ imageUrl: "Image URL", title: "Image 2" })
        ],
        onCarouselSlideChange: jest.fn()
      })
      .mount();

    it("navigates to the second image", () => {
      wrapper
        .find("Carousel Button.Carousel--pagination--page")
        .at(1)
        .simulate("click");
    });

    it("calls the onCarouselSlideChange prop with the expected payload", () => {
      expect(props.onCarouselSlideChange).toHaveBeenCalledTimes(1);
      expect(props.onCarouselSlideChange).toHaveBeenCalledWith(1);
    });

    it("sets the onCarouselSlideChange prop to be undefined", () => {
      wrapper.setProps({ onCarouselSlideChange: undefined });
    });

    it("navigates to the first image", () => {
      wrapper
        .find("Carousel Button.Carousel--pagination--page")
        .at(0)
        .simulate("click");
    });
  });
});
