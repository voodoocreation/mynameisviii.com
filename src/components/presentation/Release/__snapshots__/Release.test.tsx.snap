// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[presentation] <Release /> when all props are defined, with multiple discs matches snapshot 1`] = `
<article
  class="Release"
>
  <header
    class="PageHeader"
  >
    <h1>
      Title
    </h1>
  </header>
  <div
    class="MetaBar Release--meta"
  >
    <div
      class="Meta Release--type"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 14.5c-2.49 0-4.5-2.01-4.5-4.5S9.51 7.5 12 7.5s4.5 2.01 4.5 4.5-2.01 4.5-4.5 4.5zm0-5.5c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Type:
      </span>
       
      <span
        class="Meta--value"
      >
        Album
      </span>
    </div>
    <div
      class="Meta Release--releasedOn"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Released:
      </span>
       
      <span
        class="Meta--value"
      >
        <time
          class="DateTime"
          datetime="2017-01-01"
          title="1 January 2017"
        >
          1 January 2017
        </time>
      </span>
    </div>
    <div
      class="Meta Release--length"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Length:
      </span>
       
      <span
        class="Meta--value"
      >
        5:00
      </span>
    </div>
    <div
      class="Meta Release--genre"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3v10.55c-.59-.34-1.27-.55-2-.55-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4V7h4V3h-6z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Genre:
      </span>
       
      <span
        class="Meta--value"
      >
        Genre
      </span>
    </div>
    <div
      class="Meta Release--tracks"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 17h2v.5H3v1h1v.5H2v1h3v-4H2v1zm1-9h1V4H2v1h1v3zm-1 3h1.8L2 13.1v.9h3v-1H3.2L5 10.9V10H2v1zm5-6v2h14V5H7zm0 14h14v-2H7v2zm0-6h14v-2H7v2z"
        />
      </svg>
        
      <span
        class="Meta--value"
      >
        4 tracks
      </span>
    </div>
  </div>
  <div
    class="Release--body"
  >
    <section
      class="Release--images"
    >
      <div
        class="Carousel"
      >
        <div
          class="Carousel--slides"
          style="transform:translate3d(-0%, 0, 0)"
        >
          <figure
            class="Image Release--images--image isLoading"
          >
            <div
              class="Image--wrapper"
            >
              <img
                alt="Image title"
                src="Image URL"
              />
              <span
                class="Loader"
              >
                <div
                  class="Loader--animation"
                >
                  <span />
                  <span />
                  <span />
                </div>
              </span>
            </div>
            <figcaption>
              Image title
            </figcaption>
          </figure>
        </div>
      </div>
    </section>
    <div
      class="Release--details"
    >
      <section
        class="Release--description"
      >
        Description
      </section>
      <section
        class="Release--tracklist"
      >
        <h2>
          Tracklist
        </h2>
        <h3>
          Disc 0
        </h3>
        <ol>
          <li>
            <span
              class="Release--tracklist--title"
            >
              Disc 1, Track 1
            </span>
             
            <span
              class="Release--tracklist--genre"
            />
             
            <span
              class="Release--tracklist--length"
            >
              0:00
            </span>
          </li>
          <li>
            <span
              class="Release--tracklist--title"
            >
              Disc 1, Track 2
            </span>
             
            <span
              class="Release--tracklist--genre"
            />
             
            <span
              class="Release--tracklist--length"
            >
              0:00
            </span>
          </li>
        </ol>
        <h3>
          Disc 1
        </h3>
        <ol>
          <li>
            <span
              class="Release--tracklist--title"
            >
              Disc 2, Track 1
            </span>
             
            <span
              class="Release--tracklist--genre"
            />
             
            <span
              class="Release--tracklist--length"
            >
              0:00
            </span>
          </li>
          <li>
            <span
              class="Release--tracklist--title"
            >
              Disc 2, Track 2
            </span>
             
            <span
              class="Release--tracklist--genre"
            />
             
            <span
              class="Release--tracklist--length"
            >
              0:00
            </span>
          </li>
        </ol>
      </section>
      <section
        class="Release--streamList"
      >
        <h2>
          Stream
        </h2>
        <ul
          class="Release--platformList"
        >
          <li>
            <a
              href="URL"
              rel="noopener noreferrer"
              target="_blank"
              title="Stream {title} on Spotify"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 496 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M248 8C111.1 8 0 119.1 0 256s111.1 248 248 248 248-111.1 248-248S384.9 8 248 8zm100.7 364.9c-4.2 0-6.8-1.3-10.7-3.6-62.4-37.6-135-39.2-206.7-24.5-3.9 1-9 2.6-11.9 2.6-9.7 0-15.8-7.7-15.8-15.8 0-10.3 6.1-15.2 13.6-16.8 81.9-18.1 165.6-16.5 237 26.2 6.1 3.9 9.7 7.4 9.7 16.5s-7.1 15.4-15.2 15.4zm26.9-65.6c-5.2 0-8.7-2.3-12.3-4.2-62.5-37-155.7-51.9-238.6-29.4-4.8 1.3-7.4 2.6-11.9 2.6-10.7 0-19.4-8.7-19.4-19.4s5.2-17.8 15.5-20.7c27.8-7.8 56.2-13.6 97.8-13.6 64.9 0 127.6 16.1 177 45.5 8.1 4.8 11.3 11 11.3 19.7-.1 10.8-8.5 19.5-19.4 19.5zm31-76.2c-5.2 0-8.4-1.3-12.9-3.9-71.2-42.5-198.5-52.7-280.9-29.7-3.6 1-8.1 2.6-12.9 2.6-13.2 0-23.3-10.3-23.3-23.6 0-13.6 8.4-21.3 17.4-23.9 35.2-10.3 74.6-15.2 117.5-15.2 73 0 149.5 15.2 205.4 47.8 7.8 4.5 12.9 10.7 12.9 22.6 0 13.6-11 23.3-23.2 23.3z"
                />
              </svg>
               Spotify
            </a>
          </li>
        </ul>
      </section>
      <section
        class="Release--buyList"
      >
        <h2>
          Buy
        </h2>
        <ul
          class="Release--platformList"
        >
          <li>
            <a
              href="URL"
              rel="noopener noreferrer"
              target="_blank"
              title="Buy {title} from iTunes"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 384 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M318.7 268.7c-.2-36.7 16.4-64.4 50-84.8-18.8-26.9-47.2-41.7-84.7-44.6-35.5-2.8-74.3 20.7-88.5 20.7-15 0-49.4-19.7-76.4-19.7C63.3 141.2 4 184.8 4 273.5q0 39.3 14.4 81.2c12.8 36.7 59 126.7 107.2 125.2 25.2-.6 43-17.9 75.8-17.9 31.8 0 48.3 17.9 76.4 17.9 48.6-.7 90.4-82.5 102.6-119.3-65.2-30.7-61.7-90-61.7-91.9zm-56.6-164.2c27.3-32.4 24.8-61.9 24-72.5-24.1 1.4-52 16.4-67.9 34.9-17.5 19.8-27.8 44.3-25.6 71.9 26.1 2 49.9-11.4 69.5-34.3z"
                />
              </svg>
               iTunes
            </a>
          </li>
        </ul>
      </section>
      <section
        class="Release--copyright"
      >
        <p>
          ℗© 2017 Record Label
        </p>
      </section>
    </div>
  </div>
  <script
    type="application/ld+json"
  >
    {"@id":"http://localhost:80/releases/release-1","@type":"MusicAlbum","albumProductionType":"http://schema.org/StudioAlbum","albumRelease":{"@type":"MusicRelease","duration":"PT5M0S","name":"Title","recordLabel":"Record Label"},"albumReleaseType":"http://schema.org/album","byArtist":{"@type":"MusicGroup","name":"Artist Name"},"datePublished":"2017-01-01T00:00:00","description":"Description","image":"Image URL","mainEntityOfPage":{"@id":"http://localhost:80/releases/release-1"},"name":"Title","track":{"@type":"ItemList","itemListElement":[{"@type":"ListItem","item":{"@type":"MusicRecording","duration":"PT0M0S","name":"Disc 1, Track 1","recordingOf":{"@type":"MusicComposition","name":"Disc 1, Track 1"}},"position":1},{"@type":"ListItem","item":{"@type":"MusicRecording","duration":"PT0M0S","name":"Disc 1, Track 2","recordingOf":{"@type":"MusicComposition","name":"Disc 1, Track 2"}},"position":2},{"@type":"ListItem","item":{"@type":"MusicRecording","duration":"PT0M0S","name":"Disc 2, Track 1","recordingOf":{"@type":"MusicComposition","name":"Disc 2, Track 1"}},"position":3},{"@type":"ListItem","item":{"@type":"MusicRecording","duration":"PT0M0S","name":"Disc 2, Track 2","recordingOf":{"@type":"MusicComposition","name":"Disc 2, Track 2"}},"position":4}],"numberOfItems":4},"@context":"http://schema.org/"}
  </script>
</article>
`;

exports[`[presentation] <Release /> when the minimum props are defined, with only one disc matches snapshot 1`] = `
<article
  class="Release"
>
  <header
    class="PageHeader"
  >
    <h1>
      Title
    </h1>
  </header>
  <div
    class="MetaBar Release--meta"
  >
    <div
      class="Meta Release--type"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 14.5c-2.49 0-4.5-2.01-4.5-4.5S9.51 7.5 12 7.5s4.5 2.01 4.5 4.5-2.01 4.5-4.5 4.5zm0-5.5c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Type:
      </span>
       
      <span
        class="Meta--value"
      >
        Album
      </span>
    </div>
    <div
      class="Meta Release--releasedOn"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Released:
      </span>
       
      <span
        class="Meta--value"
      >
        <time
          class="DateTime"
          datetime="2017-01-01"
          title="1 January 2017"
        >
          1 January 2017
        </time>
      </span>
    </div>
    <div
      class="Meta Release--length"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Length:
      </span>
       
      <span
        class="Meta--value"
      >
        5:00
      </span>
    </div>
    <div
      class="Meta Release--genre"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3v10.55c-.59-.34-1.27-.55-2-.55-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4V7h4V3h-6z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Genre:
      </span>
       
      <span
        class="Meta--value"
      >
        Genre
      </span>
    </div>
    <div
      class="Meta Release--tracks"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 17h2v.5H3v1h1v.5H2v1h3v-4H2v1zm1-9h1V4H2v1h1v3zm-1 3h1.8L2 13.1v.9h3v-1H3.2L5 10.9V10H2v1zm5-6v2h14V5H7zm0 14h14v-2H7v2zm0-6h14v-2H7v2z"
        />
      </svg>
        
      <span
        class="Meta--value"
      >
        1 track
      </span>
    </div>
  </div>
  <div
    class="Release--body"
  >
    <section
      class="Release--images"
    >
      <div
        class="Carousel"
      >
        <div
          class="Carousel--slides"
          style="transform:translate3d(-0%, 0, 0)"
        >
          <figure
            class="Image Release--images--image isLoading"
          >
            <div
              class="Image--wrapper"
            >
              <img
                alt="Image title"
                src="Image URL"
              />
              <span
                class="Loader"
              >
                <div
                  class="Loader--animation"
                >
                  <span />
                  <span />
                  <span />
                </div>
              </span>
            </div>
            <figcaption>
              Image title
            </figcaption>
          </figure>
        </div>
      </div>
    </section>
    <div
      class="Release--details"
    >
      <section
        class="Release--description"
      >
        Description
      </section>
      <section
        class="Release--tracklist"
      >
        <h2>
          Tracklist
        </h2>
        <ol>
          <li>
            <span
              class="Release--tracklist--title"
            >
              Title
            </span>
             
            <span
              class="Release--tracklist--genre"
            >
              Genre
            </span>
             
            <span
              class="Release--tracklist--length"
            >
              5:00
            </span>
          </li>
        </ol>
      </section>
      <section
        class="Release--copyright"
      >
        <p>
          ℗© 2017 Record Label
        </p>
      </section>
    </div>
  </div>
  <script
    type="application/ld+json"
  >
    {"@id":"http://localhost:80/releases/release-1","@type":"MusicAlbum","albumProductionType":"http://schema.org/StudioAlbum","albumRelease":{"@type":"MusicRelease","duration":"PT5M0S","name":"Title","recordLabel":"Record Label"},"albumReleaseType":"http://schema.org/album","byArtist":{"@type":"MusicGroup","name":"Artist Name"},"datePublished":"2017-01-01T00:00:00","description":"Description","image":"Image URL","mainEntityOfPage":{"@id":"http://localhost:80/releases/release-1"},"name":"Title","track":{"@type":"ItemList","itemListElement":[{"@type":"ListItem","item":{"@type":"MusicRecording","duration":"PT5M0S","name":"Title","recordingOf":{"@type":"MusicComposition","name":"Title"}},"position":1}],"numberOfItems":1},"@context":"http://schema.org/"}
  </script>
</article>
`;
